compiler/ballerina-lang/src/main/resources/compiler.properties

```
@@ -1881,3 +1881,6 @@ error.constant.cyclic.reference=\
 
 error.unsupported.multilevel.closures=\
     closure variable ''{0}'' : closures not yet supported for object constructors which are fields
+
+error.infer.size.only.supported.in.the.first.dimension=\
+  invalid usage of closed type: infer array size only supported in the first dimension
```

IMO closed concept is not available for arrays/lists in the specification

---

@MaryamZi :help:

---

Yeah, we used to call arrays open and closed at one point, but we don't anymore. I think this can just be
```suggestion
  inferred array size is only allowed in the first dimension of an array type descriptor
```